File: ghc-4335c07/nofib/real/gamteb/inLML/gamtebType.m
-- 
--      Patricia Fasel
--      Los Alamos National Laboratory
--      1990 August
--
module

#define Angle		Float
#define	Count		Int
#define Energy		Float
#define	Indx		Int
#define	PartType	Int
#define	Random		Float
#define	StatType	Int
#define Weight		Float
#define Value		Float

export Point, Particle, Prob, Result, Stat;

rec
type Point = Pt Float Float Float

and
type Particle =
	Part
	Point			-- position	x,y,z
	Point			-- dir_cosine	u,v,w
	Weight			-- weight	wt
	Energy			-- energy	erg
	Indx			-- e_indx	
	Int			-- cell		ia
	Random			-- seed		ps

and
type Prob =
	Pr
	Float			-- pcompton
	Float			-- ppair
	Float			-- pphoto
	Float			-- ptotal

and
type Result =
	Res
	PartType		-- did particle scatter, escape or transit
	Indx			-- index of final energy in energy table
	Weight			-- final weight of particle
	
and
type Stat =
	St
	StatType		-- which statistic is incremented
	Value			-- increment value

end

File: ghc-4335c07/nofib/real/gamteb/GamtebType.hs
-- 
--      Patricia Fasel
--      Los Alamos National Laboratory
--	GAMTEB: Monte Carlo
--      1990 August
--
module GamtebType (Angle, Coord, Energy, Indx, PartType, Prob, 
		 Random, StatType, Weight, Value, Result, Stat,
		 Point, Particle(..), Probability) where

type	Angle		= Double
type	Coord		= Double
type	Energy		= Double
type	Indx		= Int		-- index into energy and xsect tables
type	PartType	= Int		-- particle scatter, escape, transmit
type	Prob		= Double
type	Random		= Double
type	StatType	= Int		-- which statistic is incremented
type	Weight		= Double	-- final weight of the particle
type	Value		= Double
type	Result		= ((PartType, Indx), Weight)
type	Stat		= (StatType, Value)
type	Point		= (Coord, Coord, Coord)
type	Probability	= (Prob, Prob, Prob, Prob)

data Particle = 
	Part
	Point			-- position of particle
	Point			-- direction particle is moving in
	Weight			-- weight of particle
	Energy			-- energy of particle
	Indx			-- energy index of particle in table
	Int			-- cell indicating split or roulette
	Random			-- seed of random number generator
